attributes/attributes.go

```diff
@@ -99,3 +104,22 @@ func (a *Attributes) Equal(o *Attributes) bool {
 	}
 	return true
 }
+
+// String prints the attribute map. If any key or values throughout the map
+// implement fmt.Stringer, it calls that method and appends.
+func (a *Attributes) String() string {
+	var sb strings.Builder
+	sb.WriteString("{")
+	for k, v := range a.m {
+		var key, val string
+		if str, ok := k.(interface{ String() string }); ok {
+			key = str.String()
+		}
+		if str, ok := v.(interface{ String() string }); ok {
+			val = str.String()
+		}
+		sb.WriteString(fmt.Sprintf("\"%v\": \"%v\",", key, val))
```

Comment:
Does this work?
`sb.WriteString(fmt.Sprintf("%q: %q,", key, val))`

Reply:
Optional: Would be nice to not have the trailing comma after the last key-value pair. 

You would have to keep track of the first one, and for every thing else you would add a leading comma (and a space).

Reply:
Switched. Still need to get to your trailing comma.

Reply:
Should be minor. Could you get it please. Since it will help all the logs where attributes are printed out.

And when you get to it, please also add a space between pairs. So, its `"k1": "v1", "k2": "v2"` and not `"k1": "v1","k2": "v2"`

Reply:
Yeah, sounds good.

Reply:
Yeah, since non deterministic in map iteration, doesn't matter if use first one and make it last one without comma. Did that algorithm.

Reply:
Pushed. It's a lot of repeated code, I tried simplifying it but the logic really needs the extra codeblock even though it's very similar. Let me know if you can see a cleaner way.